akka-persistence-testkit/src/main/scala/akka/persistence/testkit/scaladsl/TestOps.scala

```diff
@@ -0,0 +1,350 @@
+      assert(actual == expected, s"Failed to persist $msg, got $actual instead")
+      actual
+    }, max = max.dilated, interval = pollInterval)
+
+    setIndex(persistenceId, nextInd + 1)
+    res.get.asInstanceOf[A]
+  }
+
+  /**
+   * Check that next persisted in storage for particular persistence id event matches partial function `pf`.
+   */
+  def expectNextPersistedPF[A](persistenceId: String)(pf: PartialFunction[Any, A]): A =
+    expectNextPersistedPF(persistenceId, maxTimeout)(pf)
+
+  /**
+   * Check that next persisted in storage for particular persistence id event matches partial function `pf`.
+   */
+  def expectNextPersistedPF[A](persistenceId: String, hint: String)(pf: PartialFunction[Any, A]): A =
+    expectNextPersistedPF(persistenceId, maxTimeout, hint)(pf)
+
+  /**
+   * Check for `max` time that next persisted in storage for particular persistence id event matches partial function `pf`.
+   */
+  def expectNextPersistedPF[A](persistenceId: String, max: FiniteDuration, hint: String = "")(
+      pf: PartialFunction[Any, A]): A = {
+    val nextInd = nextIndex(persistenceId)
+    val res = awaitAssert({
+      val actual = storage.findOneByIndex(persistenceId, nextInd).map(reprToAny)
+      assert(actual.isDefined, s"Expected: $hint but got unexpected None")
+      val a = actual.get
+      assert(pf.isDefinedAt(a), s"Expected: $hint but got unexpected $a")
+      a
+    }, max.dilated, interval = pollInterval)
+
+    setIndex(persistenceId, nextInd + 1)
+    pf(res)
+  }
+
+  /**
+   * Check that nothing was persisted in storage for particular persistence id.
+   */
+  def expectNothingPersisted(persistenceId: String): Unit =
+    expectNothingPersisted(persistenceId, maxTimeout)
+
+  /**
+   * Check for `max` time that nothing was persisted in storage for particular persistence id.
+   */
+  def expectNothingPersisted(persistenceId: String, max: FiniteDuration): Unit = {
+    val nextInd = nextIndex(persistenceId)
+    assertForDuration({
+      val actual = storage.findOneByIndex(persistenceId, nextInd).map(reprToAny)
+      val res = actual.isEmpty
+      assert(res, s"Found persisted event $actual, but expected None instead")
+    }, max = max.dilated, interval = pollInterval)
+  }
+
+  /**
+   * Check for `max` time that `msgs` events have been persisted in the storage in order.
+   */
+  def expectPersistedInOrder[A](
+      persistenceId: String,
+      msgs: immutable.Seq[A],
+      max: FiniteDuration): immutable.Seq[A] = {
+    val nextInd = nextIndex(persistenceId)
+    val res = awaitAssert(
+      {
+        val actual = storage.findMany(persistenceId, nextInd, msgs.size)
+        actual match {
+          case Some(reprs) =>
+            val ls = reprs.map(reprToAny)
+            assert(
+              ls.size == msgs.size && ls.zip(msgs).forall(e => e._1 == e._2),
+              "Persisted events do not correspond to expected ones")
```

Comment:
include more info about the expected and actual events, otherwise difficult to see what's wrong